plugin/src/main/java/org/wildfly/plugin/cli/ExecuteCommandsMojo.java

```diff
@@ -121,24 +122,46 @@
     /**
      * Indicates whether or not subsequent commands should be executed if an error occurs executing a command. A value of
      * {@code false} will continue processing commands even if a previous command execution results in a failure.
+     * <p>
+     * Note that this value is ignored if {@code offline} is set to {@code true}.
+     * </p>
      */
     @Parameter(alias = "fail-on-error", defaultValue = "true", property = PropertyNames.FAIL_ON_ERROR)
     private boolean failOnError = true;
 
     /**
-     * Indicates weather or not CLI should be executed in offline mode. This is useful for embedded server / host controller scripts
+     * Indicates whether or not CLI scrips or commands should be executed in an offline mode. This is useful for using
+     * an embedded server or host controller.
+     *
+     * <p>This does not start an embedded server it instead skips checking if a server is running.</p>
      */
     @Parameter(name = "offline", defaultValue = "false", property = PropertyNames.OFFLINE)
     private boolean offline = false;
 
-    @Parameter(readonly = true, defaultValue = "${project}")
-    private MavenProject project;
-
+    /**
+     * Indicates how {@code stdout} and {@code stderr} should be handled for the spawned CLI process. Currently a new
+     * process is only spawned if {@code offline} is set to {@code true}. Note that {@code stderr} will be redirected to
+     * {@code stdout} if the value is defined unless the value is {@code none}.
+     * <div>
+     * By default {@code stdout} and {@code stderr} are inherited from the current process. You can change the setting
+     * to one of the follow:
+     * <ul>
+     * <li>{@code none} indicates the {@code stdout} and {@code stderr} stream should not be consumed</li>
+     * <li>{@code System.out} or {@code System.err} to redirect to the current processes <em>(use this option if you
+     * see odd behavior from maven with the default value)</em></li>
+     * <li>Any other value is assumed to be the path to a file and the {@code stdout} and {@code stderr} will be
+     * written there</li>
+     * </ul>
+     * </div>
+     */
+    @Parameter(name = "stdout", defaultValue = "System.out", property = PropertyNames.STDOUT)
```

Comment:
I think for now I'm just going to leave it as `System.out` because it's the same default behavior for non-offline output. However making the default something like `${project.basedir}/target/wildfly-plugin/script-out.log` may make sense.